app/controllers/users/noticed_notifications_controller.rb

```diff
@@ -11,8 +11,14 @@ def index
   def mark_as_read
     notification = NoticedNotification.find(params[:notification_id])
     notification.update(read_at: Time.zone.now)
-    project = notification.params[:project]
-    redirect_to user_project_path(project.author, project)
+    if notification.type == "NewAssignmentNotification"
+      assignment = notification.params[:assignment]
+      group = assignment.group
+      redirect_to group_assignment_path(group, assignment)
+    else
+      project = notification.params[:project]
+      redirect_to user_project_path(project.author, project)
```

Comment:
We need to clean this up and move logic to a service [object](https://www.toptal.com/ruby-on-rails/rails-service-objects-tutorial)

Reply:
@tachyons I didn't quite get why do we need to put this in a service class? Here we are dealing with params, routing redirects ,.. controller specific things. Just as mentioned in the article you have shared.

When Should I Not Use a Service Object?
This one’s easy. I have these rules:

Does your code handle routing, params or do other controller-y things?
If so, don’t use a service object—your code belongs in the controller.

Are you trying to share your code in different controllers?
In this case, don’t use a service object—use a concern.

Is your code like a model that doesn’t need persistence?
If so, don’t use a service object. Use a non-ActiveRecord model instead.

Is your code a specific business action? (e.g., “Take out the trash,” “Generate a PDF using this text,” or “Calculate the customs duty using these complicated rules”)
In this case, use a service object. That code probably doesn’t logically fit in either your controller or your model.

Reply:
Basically notifications have several types, we are currently handling only two. Handling all those in this way will make the controller too big. We need to tidy this up

Reply:
@VaibhavUpreti This will help in clear code!